libs/net/nodenetworkconfigurationpolicy.py

```diff
@@ -0,0 +1,183 @@
+            node_selector (dict, optional): A node selector that specifies the nodes to apply the node network
+                configuration policy to.
+        """
+        super().__init__(
+            name=name,
+            client=client,
+        )
+        self._desired_state = desired_state
+        self._node_selector = node_selector
+
+    @retry(
+        wait_timeout=TIMEOUT_90SEC,
+        sleep=TIMEOUT_5SEC,
+    )
+    def wait_for_status_success(self) -> list[dict]:
+        failed_condition_reason = self.Conditions.Reason.FAILED_TO_CONFIGURE
+        no_match_node_condition_reason = self.Conditions.Reason.NO_MATCHING_NODE
+        nncp_conditions = self.instance.status.conditions
+        for nncp_condition in nncp_conditions:
+            if nncp_condition["status"] == f"{True}":
+                if (
+                    nncp_condition["reason"] == self.Conditions.Reason.SUCCESSFULLY_CONFIGURED
+                    and nncp_condition["status"] == self.Condition.Status.TRUE
+                ):
+                    self.logger.info(f"{self.kind} {self.name} configured successfully")
+                    return nncp_condition
+
+                elif (
+                    nncp_condition["reason"] == no_match_node_condition_reason
+                    or nncp_condition["reason"] == failed_condition_reason
+                ):
+                    raise NNCPConfigurationFailed(
+                        f"{self.kind} {self.name} failed to configure with error: {nncp_condition['reason']}\n"
+                    )
+
+        raise NNCPConfigurationFailed(f"{self.name}. None of the conditions were met.")
+
+    def deploy(self, wait: bool = False) -> NodeNetworkConfigurationPolicy:
+        self.create(wait=wait)
+        try:
+            self.wait_for_status_success()
+            return self
+        except Exception as exp:
+            self.logger.error(exp)
+            super().__exit__()
+            raise
+
+    def to_dict(self) -> None:
+        super().to_dict()
+        if not self.kind_dict and not self.yaml_file:
+            self.res.setdefault("spec", {}).update({
+                "desiredState": asdict(self._desired_state, dict_factory=dict_normalization),
+                "nodeSelector": self._node_selector if self._node_selector else {},
+            })
+
+    def _delete_interfaces(self) -> None:
+        desired_state = {}
+        if self._desired_state.interfaces:
+            for iface in self._desired_state.interfaces:
+                iface.state = Resource.Interface.State.ABSENT
+
+            desired_state["interfaces"] = [asdict(iface) for iface in self._desired_state.interfaces]
+
+        ResourceEditor(patches={self: {"spec": {"desiredState": desired_state}}}).update()
+
+    def clean_up(self) -> bool:
+        try:
+            self._delete_interfaces()
+            self.wait_for_status_success()
+        except Exception as exp:
+            self.logger.error(exp)
+
+        return self.delete()
```

Comment:
Why it is not deleted through the context manager? Why would we want to remove the object through special methods?

Reply:
Related to prev thread: https://github.com/RedHatQE/openshift-virtualization-tests/pull/218/files#r1954511796